libs/tx-builder-feature/src/utils/search.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  ABI,
+  ArbitraryState,
+  isArgType,
+  JSONDetailsSearch,
+  StringSearch,
+  ValidNetwork,
+} from '@daohaus/common-utilities';
+import { processArg } from './args';
+
+export const checkArgType = (arg: unknown) => {
+  if (isArgType(arg)) {
+    return arg;
+  }
+  throw new Error(`Invalid arg type ${arg}`);
+};
+
+export const deepSearch = (
```

Comment:
i really like the name of this function.

![Uploading Screenshot from 2022-08-15 08-39-29.png…]()

Reply:
haha